overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -327,187 +772,158 @@ create_dump()
 #
 setup_datasets()
 {
-  datasets=$(zfs list -H -o name | xargs)
-
-  if ! echo $datasets | grep dump > /dev/null; then
-    printf "%-56s" "Making dump zvol... "
-    create_dump
-    printf "%4s\n" "done"
-  fi
-
-  if ! echo $datasets | grep ${CONFDS} > /dev/null; then
-    printf "%-56s" "Initializing config dataset for zones... "
-    zfs create ${CONFDS} || fatal "failed to create the config dataset"
-    chmod 755 /${CONFDS}
-    cp -p /etc/zones/* /${CONFDS}
-    zfs set mountpoint=legacy ${CONFDS}
-    printf "%4s\n" "done"
-  fi
-
-  if ! echo $datasets | grep ${USBKEYDS} > /dev/null; then
-    if [[ -n $(/bin/bootparams | grep "^smartos=true") ]]; then
-        printf "%-56s" "Creating config dataset... "
-        zfs create -o mountpoint=legacy ${USBKEYDS} || \
-          fatal "failed to create the config dataset"
-        mkdir /usbkey
-        mount -F zfs ${USBKEYDS} /usbkey
-        printf "%4s\n" "done"
-    fi
-  fi
-
-  if ! echo $datasets | grep ${COREDS} > /dev/null; then
-    printf "%-56s" "Creating global cores dataset... "
-    zfs create -o quota=10g -o mountpoint=/${SYS_ZPOOL}/global/cores \
-        -o compression=gzip ${COREDS} || \
-        fatal "failed to create the cores dataset"
-    printf "%4s\n" "done"
-  fi
-
-  if ! echo $datasets | grep ${OPTDS} > /dev/null; then
-    printf "%-56s" "Creating opt dataset... "
-    zfs create -o mountpoint=legacy ${OPTDS} || \
-      fatal "failed to create the opt dataset"
-    printf "%4s\n" "done"
-  fi
-
-  if ! echo $datasets | grep ${VARDS} > /dev/null; then
-    printf "%-56s" "Initializing var dataset... "
-    zfs create ${VARDS} || \
-      fatal "failed to create the var dataset"
-    chmod 755 /${VARDS}
-    cd /var
-    if ( ! find . -print | cpio -pdm /${VARDS} 2>/dev/null ); then
-        fatal "failed to initialize the var directory"
-    fi
+	datasets=$(zfs list -H -o name | xargs)
 
-    zfs set mountpoint=legacy ${VARDS}
-
-    if ! echo $datasets | grep ${SWAPVOL} > /dev/null; then
-          printf "%-56s" "Creating swap zvol... "
-          #
-          # We cannot allow the swap size to be less than the size of DRAM, lest$
-          # we run into the availrmem double accounting issue for locked$
-          # anonymous memory that is backed by in-memory swap (which will$
-          # severely and artificially limit VM tenancy).  We will therfore not$
-          # create a swap device smaller than DRAM -- but we still allow for the$
-          # configuration variable to account for actual consumed space by using$
-          # it to set the refreservation on the swap volume if/when the$
-          # specified size is smaller than DRAM.$
-          #
-          size=${SYSINFO_MiB_of_Memory}
-          zfs create -V ${size}mb ${SWAPVOL}
-          swap -a /dev/zvol/dsk/${SWAPVOL}
-    fi
-    printf "%4s\n" "done"
-  fi
-}
+	if ! echo $datasets | grep dump > /dev/null; then
```

Comment:
this could potentially match anything with the word dump in it... wouldn't you want something tighter like


``` bash
if ! grep -q '^zones/dump$' <<< "$datasets"; then
  # make dump zvole
fi
```